src/win/fs.c

```diff
@@ -764,8 +764,64 @@ void fs__mkdir(uv_fs_t* req) {
 }
 
 
-/* OpenBSD original: lib/libc/stdio/mktemp.c */
+static void fs__mkdtemp_callback(uv_fs_t* req, static const size_t num_x, WCHAR* ep) {
+  size_t len;
+
+  if (_wmkdir(req->file.pathw) == 0) {
+    len = strlen(req->path);
+    wcstombs((char*) req->path + len - num_x, ep - num_x, num_x);
+    SET_REQ_RESULT(req, 0);
+    return 1;
+  } else if (errno != EEXIST) {
+    SET_REQ_RESULT(req, -1);
+    return 1;
+  }
+
+  return 0;
+}
+
+
 void fs__mkdtemp(uv_fs_t* req) {
+  fs__mktemp(req, fs__mkdtemp_callback);
+}
+
+
+int fs__mkstemp_callback(uv_fs_t* req, static const size_t num_x, WCHAR* ep) {
+  size_t len;
+  DWORD error;
+  int fd;
+
+  fd = CreateFileW(req->file.pathw,
```

Comment:
after getting the file handle we need to extract its "unix style" fd suing `_open_osfhandle`, have a look at how open does it on Windows.